lib/nano/nano.js

```diff
@@ -0,0 +1,76 @@
+import { blake2b } from 'blakejs'
+
+const nanoAlphabet = '13456789abcdefghijkmnopqrstuwxyz'
+
+function nanobase32(buffer) {
+  const length = buffer.length
+  const leftover = (length * 8) % 5
+  const offset = leftover === 0 ? 0 : 5 - leftover
+  let value = 0
+  let output = ''
+  let bits = 0
+  for (let i = 0; i < length; i++) {
+    value = (value << 8) | buffer[i]
+    bits += 8
+    while (bits >= 5) {
+      output += nanoAlphabet[(value >>> (bits + offset - 5)) & 31]
+      bits -= 5
+    }
+  }
+  if (bits > 0) {
+    output += nanoAlphabet[(value << (5 - (bits + offset))) & 31]
+  }
+  return output
+}
+
+export function encodeNanoAddr(data) {
+  const hex = data.toString('hex');
+  const length = (hex.length / 2) | 0;
+  const uint8 = new Uint8Array(length);
+  for (let i = 0; i < length; i++) {
+    uint8[i] = parseInt(hex.substr(i * 2, 2), 16);
+  }
+  const address = nanobase32(uint8);
+  const checksumbytes = blake2b(uint8, null, 5).reverse();
+  const checksum = nanobase32(checksumbytes);
+  return 'nano_' + address + checksum
+}
+
+export function decodeNanoAddr(account) {
+  if (
```

Comment:
Can you restructure this so the error is the default case (eg, invert the logic)? That would make it much easier to read.

Reply:
Better yet, both tests could be done with a regular expression.